### DISCUSSION TITLE
[Website] Promotional materials - Slides -> Include RavenDb vs Elasticsearch

### DISCUSSION THREAD

**Discussion by PureKrome** [ravendb/ravendb#13345] [Ideas]
Would love a slide to compare RavenDB to Elasticsearch as this is a common comparison I've stumbled across with various people over the years. It's frustrating to hear people choosing ES because they haven't heard of RavenDb and if when they do, they don't know the differences.

**Comment by ayende** [comment]
That is a great idea. What would be your pros and cons in this case? What do you think would have the biggest impact?

**Reply by PureKrome** [reply]
ES Pro's
- large community == easyish to get support
- fast
- FTS: good results, when you know what you're doing. (it is backed by Lucene, after all)
- has a docker image which is easy to use
- xplat

ES Con's
- everything else
- shit DSL. like .. urgh ....
- no dashboard (please don't tell me to use kibana. like .. URGH .......)
- srsly, no easy way to import data? Think, test cases or localhost dev (you have a pre-seeded db ... docker-compose up, etc)
- crap ass documentation

RavenDb Pro's
- lovely DSL
- option to write indexes in code or via the dashboard or even (if your YOLO ... auto indexes)
- lovely dashboard
- can import data really easily
- has a LOVELY test harness, even with a kewl easter egg! Yes people, learn about `waitForUserToContinueTheTest` plz.
- has a docker image which is easy to use
- docs are nice / usable
- playground
- xplat
- SaaS in Azure/AWS/GC
- fast
- FTS: good results (it is also backed by Lucene)
- does things other than FTS, like I can just store data is an -easy- way.
- 
RavenDb Con's
- Not known enough in the community. Many people don't know of it, still.
- Some people (in the < v4 days) have bad memories of early issues, like the 128 docs/session. See: octopus deploy removing off it, back to sql. Hard to get them to try v4/v5/etc
- tiny community, so community-support is low
- no 'popular' website(s) or products that people can reference as a 'oh! if it's used there, then it might ok for us, too'.

---
these are just some noobie things I've quickly jotted down. I've not done hardcore stuff like Clustering, 1000000000000 nodes, etc. Also, i've not played with Time series, etc. 

HTH

**Reply by ayende** [reply]
Thanks!

**Reply by ngbrown** [reply]
An ES Pro: In the past, I've directly exposed the Elasticsearch search [query string query](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-query-string-query.html) to application users enable aspect filtering from the search box. Now with RavenDB I'll have to custom roll my own solution to do the same?

**Reply by ayende** [reply]
You can expose that directly with the `lucene()` method, which allows you to send a user defined string using the `lucene` format and process that.
